[BUG] VM test cases are failing

## What's the test to develop? Please describe

The below tests related to VMs are failing.
1. integration/test_vm_functions.py::TestVMOperations::test_softreboot
```
    def test_softreboot(self, api_client, unique_vm_name, wait_timeout):
        '''
        Steps:
            1. Softreboot the VM was created
        Exepected Result:
            - VM's qemu-agent should disconnected (which means the VM rebooting)
            - VM's qemu-agent should re-connected (which means the VM boot into OS)
            - VM's status should be changed to `Running`
        '''
        code, data = api_client.vms.get_status(unique_vm_name)
        assert 200 == code, (
            f"unable to get VM({unique_vm_name})'s instance infos with errors:\n"
            f"Status({code}): {data}"
        )
        old_agent = data['status']['conditions'][-1]
        assert "AgentConnected" == old_agent['type'], (code, data)
    
        api_client.vms.softreboot(unique_vm_name)
        endtime = datetime.now() + timedelta(seconds=wait_timeout)
        while endtime > datetime.now():
            code, data = api_client.vms.get_status(unique_vm_name)
            if "AgentConnected" not in data['status']['conditions'][-1]['type']:
                break
            sleep(5)
    
        endtime = datetime.now() + timedelta(seconds=wait_timeout)
        while endtime > datetime.now():
            code, data = api_client.vms.get_status(unique_vm_name)
>           phase, conds = data['status']['phase'], data['status']['conditions']
E           KeyError: 'conditions'
```
2. integration/test_vm_functions.py::TestVMOperations::test_migrate
```
def test_migrate(self, api_client, unique_vm_name, wait_timeout):
        """
        To cover test:
        - https://harvester.github.io/tests/manual/live-migration/migrate-turned-on-vm-to-another-host/ # noqa
    
        Steps:
            1. migrate the VM was created
        Exepected Result:
            - VM's host Node should be changed to another one
        """
        code, host_data = api_client.hosts.get()
        assert 200 == code, (code, host_data)
        code, data = api_client.vms.get_status(unique_vm_name)
>       cur_host = data['status']['nodeName']
E       KeyError: 'nodeName'
```
3. integration/test_vm_functions.py::TestVMOperations::test_abort_migrate
```
def test_abort_migrate(self, api_client, unique_vm_name, wait_timeout):
        """
        To cover test:
        - https://harvester.github.io/tests/manual/live-migration/abort-live-migration/
    
        Steps:
            1. Abort the VM was created and migrating
        Exepected Result:
            - VM should able to perform migrate
            - VM should stay in current host when migrating be aborted.
        """
        code, host_data = api_client.hosts.get()
        assert 200 == code, (code, host_data)
        code, data = api_client.vms.get_status(unique_vm_name)
>       cur_host = data['status']['nodeName']
E       KeyError: 'nodeName'
```

--- a/harvester_e2e_tests/integration/test_vm_functions.py
+++ b/harvester_e2e_tests/integration/test_vm_functions.py
@@ -296,18 +296,19 @@ class TestVMOperations:
         assert "AgentConnected" == old_agent['type'], (code, data)
 
         api_client.vms.softreboot(unique_vm_name)
+        # Wait until agent disconnected (leaving OS)
         endtime = datetime.now() + timedelta(seconds=wait_timeout)
         while endtime > datetime.now():
             code, data = api_client.vms.get_status(unique_vm_name)
             if "AgentConnected" not in data['status']['conditions'][-1]['type']:
                 break
             sleep(5)
-
+        # then wait agent connected again (Entering OS)
         endtime = datetime.now() + timedelta(seconds=wait_timeout)
         while endtime > datetime.now():
             code, data = api_client.vms.get_status(unique_vm_name)
-            phase, conds = data['status']['phase'], data['status']['conditions']
-            if "Running" == phase and "AgentConnected" == conds[-1]['type']:
+            phase, conds = data['status']['phase'], data['status'].get('conditions', [{}])
+            if "Running" == phase and "AgentConnected" == conds[-1].get('type'):
                 break
             sleep(3)
         else:
@@ -338,7 +339,11 @@ class TestVMOperations:
         code, host_data = api_client.hosts.get()
         assert 200 == code, (code, host_data)
         code, data = api_client.vms.get_status(unique_vm_name)
-        cur_host = data['status']['nodeName']
+        cur_host = data['status'].get('nodeName')
+        assert cur_host, (
+            f"VMI exists but `nodeName` is empty.\n"
+            f"{data}"
+        )
 
         new_host = next(h['id'] for h in host_data['data'] if cur_host != h['id'])
 
@@ -372,7 +377,11 @@ class TestVMOperations:
         code, host_data = api_client.hosts.get()
         assert 200 == code, (code, host_data)
         code, data = api_client.vms.get_status(unique_vm_name)
-        cur_host = data['status']['nodeName']
+        cur_host = data['status'].get('nodeName')
+        assert cur_host, (
+            f"VMI exists but `nodeName` is empty.\n"
+            f"{data}"
+        )
 
         new_host = next(h['id'] for h in host_data['data'] if cur_host != h['id'])
 

